### PR TITLE
eduke32: implement proper macOS support

### PR DESCRIPTION
**Pull request by mikroskeem**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/eduke32/default.nix [hunk 1]
```diff
@@ -16,21 +17,37 @@ let
...
+  ] ++ lib.optionals stdenv.isLinux [
+    alsa-lib
+    gtk2
+    libGL
+    libGLU
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.AGL
+    darwin.apple_sdk.frameworks.Cocoa
+    darwin.apple_sdk.frameworks.GLUT
+    darwin.apple_sdk.frameworks.OpenGL
```
Comment (SuperSandro2000) [inline pkgs/games/eduke32/default.nix:44]:
```suggestion
    AGL
    Cocoa
    GLUT
    OpenGL
```

File: pkgs/games/eduke32/default.nix [hunk 2]
```diff
@@ -1,6 +1,7 @@
 { lib, stdenv, fetchurl, makeWrapper, pkg-config, nasm, makeDesktopItem
 , alsa-lib, flac, gtk2, libvorbis, libvpx, libGLU, libGL
-, SDL2, SDL2_mixer }:
+, SDL2, SDL2_mixer
+, darwin }:
```
Comment (SuperSandro2000) [inline pkgs/games/eduke32/default.nix:4]:
```suggestion
, AGL, Cocoa, GLUT, OpenGL
}:
```

**Comment by mikroskeem** [comment]
Requested changes have been made.

File: pkgs/games/eduke32/default.nix [hunk 3]
```diff
@@ -53,23 +74,33 @@ in stdenv.mkDerivation rec {
 
     install -Dm755 -t $out/bin eduke32 mapster32
 
+    runHook postInstall
```
Comment (SuperSandro2000) [inline pkgs/games/eduke32/default.nix:77]:
```suggestion
```

Comment (mikroskeem) [reply pkgs/games/eduke32/default.nix]: Whoops, fixed.

**Pull request merged**